tooling/lsp/src/test_utils.rs

```diff
@@ -37,3 +37,23 @@ pub(crate) async fn init_lsp_server(directory: &str) -> (LspState, Url) {
 
     (state, noir_text_document)
 }
+
+pub(crate) fn search_in_file(filename: &str, search_string: &str) -> Vec<Range> {
+    let file_contents = std::fs::read_to_string(filename)
+        .unwrap_or_else(|_| panic!("Couldn't read file {}", filename));
+    let file_lines: Vec<&str> = file_contents.lines().collect();
+    file_lines
+        .iter()
+        .enumerate()
+        .filter_map(|(line_num, line)| {
+            line.find(search_string).map(|index| {
```

Comment:
```suggestion
            // Note: this only finds the first instance of `search_string` on this line.
            line.find(search_string).map(|index| {
```

No need to fix this but good to flag this up imo.